pkgs/by-name/sw/switch-to-configuration-ng/src/src/main.rs

```diff
@@ -1293,7 +1293,8 @@ won't take effect until you reboot the system.
                 units_to_reload.insert(unit.to_string(), ());
                 record_unit(RELOAD_LIST_FILE, &unit)
             }
-        } else {
+        // systemd creates a transient mount unit for /nix/store, that we shouldn't try to unmount
+        } else if !matches!(mountpoint.as_str(), "/nix/store") {
```

Comment:
I don't know. It wasn't needed when we introduced the test. Was the perl version taking transient mount units into account?

Reply:
I think `stc` shouldn't touch _any_ transient units.  Same reason why transient units have `IgnoreOnIsolate=yes`.  Otherwise you'll kill things like people's graphical applications in GNOME and kill their docker containers

Reply:
Probably get_active_units should skip those units entirely?

Reply:
The dbus interface doesn't seem to return `IgnoreOnIsolate` directly for the dbus call used, but it may be accessible through some other command.

Reply:
Same as IgnoreOnIsolate, it's `GetUnit()` in https://www.freedesktop.org/software/systemd/man/latest/org.freedesktop.systemd1.html
so additional dbus call is needed. 
It's just now clear to me how to do this with the generated dbus interface:
https://gist.github.com/Mic92/37f1789f9bfc91fa7376845442fa6d80

Reply:
I don't feel familiar enough with this library. Closing this pull request instead. Probably a release blocker if it indeed can break automounts and docker container.

Reply:
>  Otherwise you'll kill things like people's graphical applications in GNOME

Won't those be user units, and thus unaffected?